HaikuPorter/BuildPlatform.py

```diff
@@ -72,7 +72,15 @@ def resolveDependencies(self, dependencyInfoFiles, requiresTypes,
 			return
 		resolver = DependencyResolver(self, requiresTypes, repositories,
 									  **kwargs)
-		return resolver.determineRequiredPackagesFor(dependencyInfoFiles)
+		resolver.determineRequiredPackagesFor(dependencyInfoFiles)
+		if Configuration.shallUseCcache():
+		    resolver.injectDependency('Required by configuration', 'cmd:ccache')
+		if Configuration.shallUseDistcc():
+		    # pump needs its own switch and is out of my scope.
+		    # see https://preview.tinyurl.com/y32dmfrv -- possible breakage!
```

Comment:
We really can't guarantee that sources/headers don't change during the build like that apparently requires, as there are lots of packages which do that.

Reply:
Yes. cmd:pump is the thing that really needs that to be true, so I didn't hook that up (and isn't functional without python3). There is still a chance that an overly-clever build script might try to invoke pump on its own volition, though, which is the scary part of that. I don't have an example of this happening, but compare to WebKit, which will detect and automatically use ld.gold if it is available (which would make linking fail on haikuwebkit, but doesn't exist right now so isn't specifically an issue)